src/transformers/models/layoutlmv3/modeling_tf_layoutlmv3.py

```diff
@@ -0,0 +1,1593 @@
+        super().__init__(**kwargs)
+        self.attention = TFLayoutLMv3Attention(config, name="attention")
+        self.intermediate = TFLayoutLMv3Intermediate(config, name="intermediate")
+        self.bert_output = TFLayoutLMv3Output(config, name="output")
+
+    def call(
+        self,
+        hidden_states: tf.Tensor,
+        attention_mask: Optional[tf.Tensor],
+        head_mask: Optional[tf.Tensor],
+        output_attentions: bool,
+        rel_pos: Optional[tf.Tensor] = None,
+        rel_2d_pos: Optional[tf.Tensor] = None,
+        training: bool = False,
+    ) -> Union[Tuple[tf.Tensor], Tuple[tf.Tensor, tf.Tensor]]:
+        self_attention_outputs = self.attention(
+            hidden_states,
+            attention_mask,
+            head_mask,
+            output_attentions=output_attentions,
+            rel_pos=rel_pos,
+            rel_2d_pos=rel_2d_pos,
+            training=training,
+        )
+        attention_output = self_attention_outputs[0]
+        outputs = self_attention_outputs[1:]  # add self attentions if we output attention weights
+        intermediate_output = self.intermediate(attention_output)
+        layer_output = self.bert_output(intermediate_output, attention_output, training=training)
+        outputs = (layer_output,) + outputs
+        return outputs
+
+
+class TFLayoutLMv3Encoder(tf.keras.layers.Layer):
+    def __init__(self, config: LayoutLMv3Config, **kwargs):
+        super().__init__(**kwargs)
+        self.config = config
+        self.layer = [TFLayoutLMv3Layer(config, name=f"layer.{i}") for i in range(config.num_hidden_layers)]
+
+        self.has_relative_attention_bias = config.has_relative_attention_bias
+        self.has_spatial_attention_bias = config.has_spatial_attention_bias
+
+        if self.has_relative_attention_bias:
+            self.rel_pos_bins = config.rel_pos_bins
+            self.max_rel_pos = config.max_rel_pos
+            self.rel_pos_bias = tf.keras.layers.Dense(
+                units=config.num_attention_heads,
+                kernel_initializer=get_initializer(config.initializer_range),
+                use_bias=False,
+                name="rel_pos_bias",
+            )
+
+        if self.has_spatial_attention_bias:
+            self.max_rel_2d_pos = config.max_rel_2d_pos
+            self.rel_2d_pos_bins = config.rel_2d_pos_bins
+            self.rel_pos_x_bias = tf.keras.layers.Dense(
+                units=config.num_attention_heads,
+                kernel_initializer=get_initializer(config.initializer_range),
+                use_bias=False,
+                name="rel_pos_x_bias",
+            )
+            self.rel_pos_y_bias = tf.keras.layers.Dense(
+                units=config.num_attention_heads,
+                kernel_initializer=get_initializer(config.initializer_range),
+                use_bias=False,
+                name="rel_pos_y_bias",
+            )
+
+    def relative_position_matrix(self, position_ids: tf.Tensor):
```

Comment:
Maybe remove one-liner function and add the line directly where it's used - helps readability a bit :-)